### PR TITLE
Adds support for Nano S+

### PR DESCRIPTION
**Pull request by t-nelson**
* Bumps target `ledger-app-builder` image to [73c9e07](https://github.com/LedgerHQ/ledger-app-builder/commit/73c9e07775ac39e178848c10b6b774b71fd3145f)
* Improves tooling support and docs for multi-device building
* Sync Makefile with current boilerplate app
* Bump Solana/SPL crates for tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,155 +1,143 @@
...
-include $(BOLOS_SDK)/Makefile.defines
 
-APPVERSION_M=1
-APPVERSION_N=2
-APPVERSION_P=0
-APPVERSION=$(APPVERSION_M).$(APPVERSION_N).$(APPVERSION_P)
-APPNAME = "Solana"
+include $(BOLOS_SDK)/Makefile.defines
 
-APP_LOAD_PARAMS = --curve ed25519 --path "44'/501'" --appFlags 0x240 $(COMMON_LOAD_PARAMS)
```
Comment (fbeutin-ledger) [inline Makefile:29]: Flag 0x40 will be dropped, no adverse side effect?

**Review by fbeutin-ledger — COMMENTED** [review]
I have some questions about the current test architecture
If I understand correctly there is no E2E automated tests, only unit tests and manual tests using a plugged device with blind-signing enabled?

Do you have an estimation of the effort needed to use Speculos instead of a plugged device?

**Comment by fbeutin-ledger** [comment]
You may also be interested in this PR https://github.com/LedgerHQ/app-solana/pull/35
Github doesn't let me tag you in it

File: Makefile (hunk 1 above)
Comment (t-nelson) [reply Makefile:29]: there shouldn't be any adverse effect.  the app never queries the pin directly.  i couldn't find any justification for why we were requesting this ability in the first place

Comment (t-nelson) [reply Makefile:29]: I _did_ fix the incorrect use of assign vs. append when specifying `--appFlags` though :wink:

File: Makefile [hunk 2]
```diff
@@ -1,155 +1,143 @@
...
-dep/%.d: %.c Makefile
+WITH_LIBSOL=1
+ifneq ($(WITH_LIBSOL),0)
+    SOURCE_FILES += $(filter-out %_test.c,$(wildcard libsol/*.c))
+    CFLAGS       += -Ilibsol/include
+    DEFINES      += HAVE_SNPRINTF_FORMAT_U
+    DEFINES      += NDEBUG
+endif
 
 load: all
```
[the remark this answers is not in the document]
Comment (t-nelson) [reply Makefile:142]: good call! for some reason i was thinking there were fall-through semantics to the next recipe if there was no empty line/recipe between rules

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**